### ISSUE TITLE
LT03/layout.operators doesn't seem to be configurable

### ISSUE THREAD

**Issue by damirbk** [sqlfluff/sqlfluff#5257]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

No matter if I choose `operator_new_lines = before` or `operator_new_lines = after`, LT03/layout.operators seems to always be set to `after`.

### Expected Behaviour

Whether SQLFluff reports linting errors or not should depend on the setting.

### Observed Behaviour

SQLFluff always reports linting errors when the binary operator is leading, and never when it is trailing.

### How to reproduce

.sqlfluff:
```
[sqlfluff]
dialect = ansi

[sqlfluff:rules:layout.operators]
operator_new_lines = before
```

test_trailing.sql:
```sql
select
    3 as col_1,
    1 +
    2 as col_2
```

test_leading.sql:
```sql
select
    3 as col_1,
    1
    + 2 as col_2
```

test_leading.sql always finishes without linting errors.
test_trailing.sql always finishes with the following linting error:
```
== [test_trailing.sql] FAIL                                                                                                                                                      
L:   3 | P:   7 | LT03 | Found trailing binary operator '+'. Expected only
                       | leading near line breaks.
                       | [layout.operators]
```
No matter if I choose `before` or `after` in .sqlfluff.

### Dialect

ANSI

### Version

SQLFluff version 2.3.2
Python version 3.10.10

### Configuration

```
[sqlfluff]
dialect = ansi

[sqlfluff:rules:layout.operators]
operator_new_lines = after
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
@damirbk - you're totally right - under the hood the rule does work, but what you've found is that the documentation has been up to date for almost 6 months since we made the configuration changes in `2.0.0`. I'll update the docs!

Thanks for catching this.

**Comment by alanmcruickshank** [comment]
@damirbk : Docs will be live soon, but you can find the correct configuration values in the linked PR (#5314).
